### PR TITLE
feat(customer): Add report to show item prices per Customer

### PR DESCRIPTION
**Pull request by Alchez**
Made to develop at #17930.

<hr>

**Reason:**

Different Customers might have different prices per Item. Having an accessible Customer Price List makes it easy to export / send for reference.

**Screenshots / GIFs:**

image

image

### CODE REVIEW

File: erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+def get_data(filters=None):
+	data = []
+	customer_details = get_customer_details(filters)
+	items = get_selling_items(filters)
+
+	for item in items:
+		price_list_rate = get_price_list_rate_for(customer_details, item.item_code) or 0.0
+		available_stock = frappe.db.sql("SELECT sum(actual_qty) FROM `tabBin` WHERE item_code = %s", item.item_code)
+		available_stock = available_stock[0][0] if available_stock else None
```
Comment (saurabh6790) [inline erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py]: Why available qty?

**Review by saurabh6790 — DISMISSED** [review]

File: erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py (hunk 1 above)
Comment (Alchez) [reply erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py]: It helps sales reps in the field to quickly pull up a customer's price list and see if that item is available for order.

File: erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		}
+	]
+
+
+def get_data(filters=None):
+	data = []
+	customer_details = get_customer_details(filters)
+	items = get_selling_items(filters)
+
+	for item in items:
```
Comment (Anurag810) [inline erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py]: Don't use query inside the loop. Guess what will happen for 1000000000+ items.
Try to optimize that.
Suggestion:
First, get all the item code at once and then try to use IN operator of SQL.

File: erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{
+ "add_total_row": 0,
```
Comment (Anurag810) [inline erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.json]: set :```"add_total_row": 0,```

**Review by Anurag810 — DISMISSED** [review]

File: erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py (hunk 2 above)
Comment (Alchez) [reply erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py]: Fixed the query.

File: erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.json (hunk 3 above)
Comment (Alchez) [reply erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.json]: Why though? The sum of rates or stock is not meaningful.

**Pull request merged**